File: csharp-dotnet2/src/main/CsharpDotNet2/Org/OpenAPITools/Model/BasicStreamAllOf.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace Org.OpenAPITools.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class BasicStreamAllOf {
    /// <summary>
    /// Gets or Sets StreamId
    /// </summary>
    [DataMember(Name="stream_id", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "stream_id")]
    public Object StreamId { get; set; }

    /// <summary>
    /// Gets or Sets Name
    /// </summary>
    [DataMember(Name="name", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "name")]
    public Object Name { get; set; }

    /// <summary>
    /// Gets or Sets Description
    /// </summary>
    [DataMember(Name="description", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "description")]
    public Object Description { get; set; }

    /// <summary>
    /// Gets or Sets DateCreated
    /// </summary>
    [DataMember(Name="date_created", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "date_created")]
    public Object DateCreated { get; set; }

    /// <summary>
    /// Gets or Sets InviteOnly
    /// </summary>
    [DataMember(Name="invite_only", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "invite_only")]
    public Object InviteOnly { get; set; }

    /// <summary>
    /// Gets or Sets RenderedDescription
    /// </summary>
    [DataMember(Name="rendered_description", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "rendered_description")]
    public Object RenderedDescription { get; set; }

    /// <summary>
    /// Gets or Sets IsWebPublic
    /// </summary>
    [DataMember(Name="is_web_public", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "is_web_public")]
    public Object IsWebPublic { get; set; }

    /// <summary>
    /// Gets or Sets StreamPostPolicy
    /// </summary>
    [DataMember(Name="stream_post_policy", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "stream_post_policy")]
    public Object StreamPostPolicy { get; set; }

    /// <summary>
    /// Gets or Sets MessageRetentionDays
    /// </summary>
    [DataMember(Name="message_retention_days", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "message_retention_days")]
    public Object MessageRetentionDays { get; set; }

    /// <summary>
    /// Gets or Sets HistoryPublicToSubscribers
    /// </summary>
    [DataMember(Name="history_public_to_subscribers", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "history_public_to_subscribers")]
    public Object HistoryPublicToSubscribers { get; set; }

    /// <summary>
    /// Gets or Sets FirstMessageId
    /// </summary>
    [DataMember(Name="first_message_id", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "first_message_id")]
    public Object FirstMessageId { get; set; }

    /// <summary>
    /// Gets or Sets IsAnnouncementOnly
    /// </summary>
    [DataMember(Name="is_announcement_only", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "is_announcement_only")]
    public Object IsAnnouncementOnly { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class BasicStreamAllOf {\n");
      sb.Append("  StreamId: ").Append(StreamId).Append("\n");
      sb.Append("  Name: ").Append(Name).Append("\n");
      sb.Append("  Description: ").Append(Description).Append("\n");
      sb.Append("  DateCreated: ").Append(DateCreated).Append("\n");
      sb.Append("  InviteOnly: ").Append(InviteOnly).Append("\n");
      sb.Append("  RenderedDescription: ").Append(RenderedDescription).Append("\n");
      sb.Append("  IsWebPublic: ").Append(IsWebPublic).Append("\n");
      sb.Append("  StreamPostPolicy: ").Append(StreamPostPolicy).Append("\n");
      sb.Append("  MessageRetentionDays: ").Append(MessageRetentionDays).Append("\n");
      sb.Append("  HistoryPublicToSubscribers: ").Append(HistoryPublicToSubscribers).Append("\n");
      sb.Append("  FirstMessageId: ").Append(FirstMessageId).Append("\n");
      sb.Append("  IsAnnouncementOnly: ").Append(IsAnnouncementOnly).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return Newtonsoft.Json.JsonConvert.SerializeObject(this, Newtonsoft.Json.Formatting.Indented);
    }

}
}

File: dart-jaguar/lib/model/basic_stream_all_of.dart
import 'package:jaguar_serializer/jaguar_serializer.dart';


part 'basic_stream_all_of.jser.dart';

class BasicStreamAllOf {
  
  @Alias('stream_id', isNullable: true,  )
  final Object streamId;
  
  @Alias('name', isNullable: true,  )
  final Object name;
  
  @Alias('description', isNullable: true,  )
  final Object description;
  
  @Alias('date_created', isNullable: true,  )
  final Object dateCreated;
  
  @Alias('invite_only', isNullable: true,  )
  final Object inviteOnly;
  
  @Alias('rendered_description', isNullable: true,  )
  final Object renderedDescription;
  
  @Alias('is_web_public', isNullable: true,  )
  final Object isWebPublic;
  
  @Alias('stream_post_policy', isNullable: true,  )
  final Object streamPostPolicy;
  
  @Alias('message_retention_days', isNullable: true,  )
  final Object messageRetentionDays;
  
  @Alias('history_public_to_subscribers', isNullable: true,  )
  final Object historyPublicToSubscribers;
  
  @Alias('first_message_id', isNullable: true,  )
  final Object firstMessageId;
  
  @Alias('is_announcement_only', isNullable: true,  )
  final Object isAnnouncementOnly;
  

  BasicStreamAllOf(
      

{
     this.streamId = null,  
     this.name = null,  
     this.description = null,  
     this.dateCreated = null,  
     this.inviteOnly = null,  
     this.renderedDescription = null,  
     this.isWebPublic = null,  
     this.streamPostPolicy = null,  
     this.messageRetentionDays = null,  
     this.historyPublicToSubscribers = null,  
     this.firstMessageId = null,  
     this.isAnnouncementOnly = null 
    
    }
  );

  @override
  String toString() {
    return 'BasicStreamAllOf[streamId=$streamId, name=$name, description=$description, dateCreated=$dateCreated, inviteOnly=$inviteOnly, renderedDescription=$renderedDescription, isWebPublic=$isWebPublic, streamPostPolicy=$streamPostPolicy, messageRetentionDays=$messageRetentionDays, historyPublicToSubscribers=$historyPublicToSubscribers, firstMessageId=$firstMessageId, isAnnouncementOnly=$isAnnouncementOnly, ]';
  }
}

@GenSerializer(nullableFields: true)
class BasicStreamAllOfSerializer extends Serializer<BasicStreamAllOf> with _$BasicStreamAllOfSerializer {

}

